test/benchgc/src/benchgc/mlir/arg.py

```diff
@@ -0,0 +1,174 @@
+
+    return mlir_args
+
+
+def get_md(tensor: torch.Tensor):
+    if tensor.ndim == 0:
+
+        class _0dMemrefDescriptor(ctypes.Structure):
+            _fields_ = [
+                ("allocated", ctypes.c_longlong),
+                ("aligned", ctypes.POINTER(dtype_to_ctype(tensor.dtype))),
+                ("offset", ctypes.c_longlong),
+            ]
+
+        md = _0dMemrefDescriptor()
+    else:
+        ctype_shape = ctypes.c_longlong * tensor.ndim
+        ctype_strides = ctypes.c_longlong * tensor.ndim
+
+        class _ndMemrefDescriptor(ctypes.Structure):
+            _fields_ = [
+                ("allocated", ctypes.c_longlong),
+                ("aligned", ctypes.POINTER(dtype_to_ctype(tensor.dtype))),
+                ("offset", ctypes.c_longlong),
+                ("shape", ctype_shape),
+                ("strides", ctype_strides),
+            ]
+
+        md = _ndMemrefDescriptor()
+        md.shape = ctype_shape(*tensor.shape)
+        md.strides = ctype_strides(*tensor.stride())
+
+    md.allocated = tensor.data_ptr()
+    md.aligned = ctypes.cast(
+        ctypes.c_void_p(tensor.data_ptr()), ctypes.POINTER(dtype_to_ctype(tensor.dtype))
+    )
+    md.offset = ctypes.c_longlong(0)
+    return md
+
+
+class MLIRArg:
+    dtype: str
+    shape: List[int]
+
+    scalar: bool
+
+    def __init__(self) -> None:
+        self.dtype = ""
+
+    # md format:
+    # 0d memref/tensor: 0xf32
+    # nd memref/tensor: 2x3xf32
+    # scalar: f32
+    def set_md(self, md: str):
+        splited: List[str] = md.split("x")
+        self.dtype = splited[-1]
+        self.shape = []
+
+        for dim in splited[:-1]:
+            self.shape.append(int(dim))
+        self.set_scalar()
+
+    def set_scalar(self):
+        # use 0xf32 to represent memref<f32>
+        # use f32 to represent f32
+        if self.shape == [0]:
+            self.shape = []
+            self.scalar = False
+        elif self.shape == []:
+            self.scalar = True
+        else:
+            self.scalar = False
+
+    def nelem(self) -> int:
+        if self.scalar or self.shape == [] or self.shape[0] == 0:
+            return 1
+        ret: int = 1
+        for dim in self.shape:
+            ret = ret * dim
+        return ret
+
+    def get_mlir_type(self, ctx: gc_mlir.ir.Context) -> gc_mlir.ir.Type:
+        if self.scalar:
+            return str_to_mlir_dtype(ctx, self.dtype)
+        else:
+            return gc_mlir.ir.RankedTensorType.get(
+                self.shape, str_to_mlir_dtype(ctx, self.dtype)
+            )
+
+    def get_ranked_tensor_type(
+        self, ctx: gc_mlir.ir.Context
+    ) -> gc_mlir.ir.RankedTensorType:
+        return gc_mlir.ir.RankedTensorType.get(
+            self.shape, str_to_mlir_dtype(ctx, self.dtype)
+        )
+
+    def get_constant_op(
+        self, ctx: gc_mlir.ir.Context, cst: Any
+    ) -> gc_mlir.dialects.tensor.OpView:
+        zero = gc_mlir.dialects.arith.ConstantOp(
```

Comment:
you can change `import gc_mlir.dialects.arith` to `from gc_mlir.dialects  import arith`, then just use `arith.ConstantOp`

Reply:
@xurui1995 
We have a package named `arith` inside `benchgc`. I think the full name path will be better to handle the ambiguous problem.